javaslang/src/main/java/javaslang/collection/JavaConverters.java

```diff
@@ -0,0 +1,437 @@
+        }
+
+        @Override
+        public ListIterator<T> listIterator() {
+            return ListIterator.of(getDelegate(), 0);
+        }
+
+        @Override
+        public ListIterator<T> listIterator(int index) {
+            return ListIterator.of(getDelegate(), index);
+        }
+
+        @Override
+        public T remove(int index) {
+            // may throw an IndexOutOfBoundsException accordingly to the j.u.List.remove(int)
+            return setDelegateAndGetPreviousElement(index, getDelegate().removeAt(index));
+        }
+
+        @Override
+        public boolean remove(Object obj) {
+            // may throw a ClassCastException accordingly to the j.u.List.remove(Object)
+            @SuppressWarnings("unchecked") final T that = (T) obj;
+            return setDelegateAndCheckChanged(getDelegate().remove(that));
+        }
+
+        @Override
+        public boolean removeAll(Collection<?> collection) {
+            // may throw a ClassCastException accordingly to the j.u.List.removeAll(Collection)
+            @SuppressWarnings("unchecked") final Collection<T> that = (Collection<T>) collection;
+            return setDelegateAndCheckChanged(getDelegate().removeAll(that));
+        }
+
+        @Override
+        public boolean retainAll(Collection<?> collection) {
+            // may throw a ClassCastException accordingly to j.u.List.retainAll(Collection)
+            @SuppressWarnings("unchecked") final Collection<T> that = (Collection<T>) collection;
+            return setDelegateAndCheckChanged(getDelegate().retainAll(that));
+        }
+
+        @Override
+        public T set(int index, T element) {
+            // may throw an IndexOutOfBoundsException accordingly to the j.u.List.set(int, T)
+            return setDelegateAndGetPreviousElement(index, getDelegate().update(index, element));
+        }
+
+        @Override
+        public int size() {
+            return getDelegate().size();
+        }
+
+        @Override
+        public List<T> subList(int fromIndex, int toIndex) {
+            // may throw IndexOutOfBoundsException accordingly to j.u.List.subList(int, int)
+            return new ListView<>(getDelegate().subSequence(fromIndex, toIndex), isMutable());
+        }
+
+        @Override
+        public Object[] toArray() {
+            return getDelegate().toJavaArray();
+        }
+
+        @Override
+        public <T1> T1[] toArray(T1[] array) {
+            return getDelegate().toJavaList().toArray(array);
+        }
+
+        // -- Object.*
+
+        @Override
+        public boolean equals(Object o) {
+            if (o == this) {
+                return true;
+            } else if (o instanceof java.util.List) {
+                return Collections.areEqual(getDelegate(), (java.util.List<?>) o);
+            } else {
+                return false;
+            }
+        }
+
+        @Override
+        public int hashCode() {
+            return getDelegate().hashCode();
```

Comment:
As I said in the other review:

I see that Collections.hash _currently_ implements the algorithm required by `java.util.List.hashCode()`, but I still think it would be safer from a maintainability point of view to use the implementation provided by AbstractList if possible _[to ensure consistency with `equals()`]_. Alternatively, Collections.hash ought to have a comment advising that the implementation must never be changed. But I don't think it would be a good idea to tie **all** the javaslang collections to follow java.util's rules forever for all. Safer just to use AbstractList. _[Assuming that's still possible with the new implementation.]_

* https://docs.oracle.com/javase/8/docs/api/java/util/List.html#hashCode--
* https://docs.oracle.com/javase/8/docs/api/java/util/AbstractList.html#hashCode--